packages/protocol/contracts/verifiers/RiscZeroVerifier.sol

```diff
@@ -76,11 +80,12 @@ contract RiscZeroVerifier is EssentialContract, IVerifier {
         );
 
         // journalDigest is the sha256 hash of the hashed public input
-        bytes32 journalDigest = sha256(bytes.concat(hash));
+        bytes32 journalDigest = sha256(bytes.concat(FIXED_JOURNAL_HEADER, hash));
 
+        emit ProofVerified(_ctx.metaHash, journalDigest, hash);
         // call risc0 verifier contract
-        (bool success,) = address(receiptVerifier).staticcall(
-            abi.encodeCall(IRiscZeroReceiptVerifier.verify, (seal, imageId, journalDigest))
+        (bool success,) = address(verifier).staticcall(
+            abi.encodeCall(IRiscZeroVerifier.verify, (seal, imageId, journalDigest))
```

Comment:
can we also add a virtual function for returning IRiscZeroVerifier, so we can override it to avoid one SLOAD?

```
funtion riskZeroVerifier() public view virtual returns (address) {
return verifier;
}
```

Reply:
added with https://github.com/taikoxyz/taiko-mono/pull/17877/commits/9481a8bb68b862fcc82387e3fd2a1998ddbec2ad